Source/path.cpp

```diff
@@ -56,6 +57,36 @@ const Displacement PathDirs[8] = {
  */
 int8_t path_directions[9] = { 5, 1, 6, 2, 0, 3, 8, 4, 7 };
 
+bool IsTileNotSolid(Point position)
+{
+	return !nSolidTable[dPiece[position.x][position.y]];

```

Comment:
I think this should do the same bounds checking as IsTileSolid to guard against OOB reads. Calling `!IsTileSolid` wouldn't be appropriate as that would return true for out of bounds positions, would need to replicate the check and return false.

Reply:
I think IsTileSolid should return true for OOB, but I would like to wait with both of these changes as there is some potential they could affecting various things at this point. So prefer getting things stable and tested first.